lib/btrfs_test.pm

```diff
@@ -40,14 +41,35 @@ way to get DBus-less environment is to enter rescue.target via systemctl.
 =cut
 sub snapper_nodbus_setup {
     my ($self) = @_;
-    if (script_run('! systemctl is-active dbus')) {
-        script_run('systemctl rescue', 0);
-        assert_screen('emergency-shell', 120);
-        type_password;
-        send_key 'ret';
-        $self->set_standard_prompt('root');
-        assert_screen 'root-console';
+    # serial console prints text style as ascii chars in PS1 by default.
+    my $escseq = '[\]\[\(\)\e\d\w\s]*';
+
+    my $rescue_service_config_path   = "/usr/lib/systemd/system/rescue.service";
+    my $rescue_service_override_path = "/etc/systemd/system/rescue.service";
+
+    return unless script_run('! systemctl is-active dbus');
+    if (!serial_terminal::select_virtio_console()) {
+        die 'Selection of virtio console failed.';
```

Comment:
I don't think it is necessary to die here. You should be able to adapt the test for both vnc and serial terminals.

Reply:
to be clear; you can probably just use the existing code if select_virtio_console returns zero. Or you can branch later with is_serial_terminal.